pkgs/misc/vscode-extensions/update_installed_exts.sh

```diff
@@ -16,39 +16,98 @@ function clean_up() {
 }
 
 function get_vsixpkg() {
-    N="$1.$2"
+    PUBLISHER=$1
+    NAME=$2
+    N="$PUBLISHER-$NAME"
 
-    # Create a tempdir for the extension download.
-    EXTTMP=$(mktemp -d -t vscode_exts_XXXXXXXX)
-
-    URL="https://$1.gallery.vsassets.io/_apis/public/gallery/publisher/$1/extension/$2/latest/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage"
-
-    # Quietly but delicately curl down the file, blowing up at the first sign of trouble.
-    curl --silent --show-error --fail -X GET -o "$EXTTMP/$N.zip" "$URL"
-    # Unpack the file we need to stdout then pull out the version
-    VER=$(jq -r '.version' <(unzip -qc "$EXTTMP/$N.zip" "extension/package.json"))
-    # Calculate the SHA
-    SHA=$(nix-hash --flat --base32 --type sha256 "$EXTTMP/$N.zip")
-
-    # Clean up.
-    rm -Rf "$EXTTMP"
-    # I don't like 'rm -Rf' lurking in my scripts but this seems appropriate.
+    USE_MS_MKTPLC="false";
+    local FETCHED_JSON_STRING="";
+    if test $# -ge 3 && test -n $3 && test $3 = "true"; then
```

Comment:
I would be happier if we use [[ instead of the old style test here.

Reply:
Would it be okay to use `[`?

Reply:
Now I see the necessity to use BASH (to propagate argument list correctly) and the advantage to use double bracket.
Suggestion applied.